core/src/main/scala/org/apache/spark/deploy/history/HistoryServerDiskManager.scala

```diff
@@ -83,6 +83,20 @@ private class HistoryServerDiskManager(
       listing.delete(info.getClass(), info.path)
     }
 
+    // Reading level db would trigger table file compaction, then it may cause size of level db
```

Comment:
We could apply "partition" instead of filter in L78, so that we only call iterate with view once and retrieve two lists (orphans, existing) - L90 can use existing to avoid calling view again.

Reply:
Thanks for your comments, updated.